QRCoder/QRCodeGenerator.ModulePlacer.BlockedModules.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections;
+
+namespace QRCoder
+{
+    public partial class QRCodeGenerator
+    {
+        private static partial class ModulePlacer
+        {
+            /// <summary>
+            /// Struct that represents blocked modules using rectangles.
+            /// </summary>
+            public struct BlockedModules : IDisposable
+            {
+                private readonly bool[,] _blockedModules;
+
+                [ThreadStatic]
```

Comment:
Is parallel execution (under load) expected?
Otherwise the thread-static could be omitted, and a plain static field be used. For rent/return `Interlocked`-methods can be used than, which is faster than the thread-static access.

This is a trade-off decision if parallel execution is very likely to happen.

Reply:
You're right, Interlocked is a better choice here.  Changed.